File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        char open = '(';
        char close = ')';

        int maxi = 0;
        int temp = 0;
        for (int i = 0; i<s.length(); i++){
            if (s.charAt(i) == open){
                temp+=1;
                if (temp>maxi){
                    maxi = temp;
                }
            }
            else if (s.charAt(i) == close){
                temp-=1;
            }
        }
    return maxi;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack<char> st;
        int maxi = 0;

        for(auto i: s){
            if(i=='('){
                st.push(i);
            }
            else if(i==')'){
                    st.pop();
            }
            maxi=max(maxi,(int)st.size());
        }
        return maxi;
    }
};